src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,47 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        validateUserParametersNotNull(user);
+        checkParameters(user);
+        checkExistingUser(user);
+
+        return storageDao.add(user);
+    }
+
+    private void validateUserParametersNotNull(User user) {
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Password can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new InvalidDataException("Age can't be null");
+        }
+    }
+
+    private void checkParameters(User user) {
+        if (user.getAge() < MIN_AGE) {
+            throw new InvalidDataException("Not valid age: "
+                    + user.getAge()
+                    + ". Min allowed age is " + MIN_AGE);
+        }
+
+        if (user.getLogin().length() < MIN_LENGTH || user.getPassword().length() < MIN_LENGTH) {
```

Comment:
you should also check whether it is not blank

Reply:
Why do I need to check whether it's not blank, if in this case it has length of 0, which if less than MIN_LENGTH anyway? Or did I get the wrong idea?

Reply:
@AlexRosenkreuz you got it wrong

empty string = `""`
blank string = `"         "` - as you can see it might have a valid length, but it is still invalid

there are 2 methods: `isEmpty` and `isBlank`

Reply:
Oh thank you. Gotta implement this